homeassistant/helpers/intent.py

```diff
@@ -66,6 +69,18 @@ def async_register(hass: HomeAssistant, handler: IntentHandler) -> None:
 
     intents[handler.intent_type] = handler
 
+    if handler.intent_type not in LLM_SKIP_AUTO_ADD_TOOL:
+        try:
+            slot_schema = handler._slot_schema  # pylint: disable=protected-access
```

Comment:
Another thought here could be: Given that the intent itself does its own validation, you could not do additional validation using the tool API. Just call `async_validate_slots` from the llm intent tool